bodhi/server/util.py

```diff
@@ -121,14 +121,14 @@ def generate_changelog(build: 'models.Build') -> typing.Optional[str]:
         return None
 
     # Grab the RPM header of the previous update, and generate a ChangeLog
-    oldh = get_rpm_header(lastpkg)
-    oldtime = oldh['changelogtime']
-    text = oldh['changelogtext']
-
-    if not text:
-        oldtime = 0
-    elif isinstance(oldtime, list):
-        oldtime = oldtime[0]
+    oldtime = 0
+    if lastpkg is not None:
+        oldh = get_rpm_header(lastpkg)
+        oldtime = oldh['changelogtime']
+        text = oldh['changelogtext']
+
+        if text and isinstance(oldtime, list):
```

Comment:
This fails in the unit tests. If `text` is falsy but `oldtime` is a list, the first element will not be extracted and the program will crash in `get_changelog()` because it does not expect a list.

Reply:
Thanks. FYI my hotfix looks like this:
```diff
diff --git a/bodhi/server/util.py b/bodhi/server/util.py
index aa611a8c5..1e2232aef 100644
--- a/bodhi/server/util.py
+++ b/bodhi/server/util.py
@@ -121,14 +121,13 @@ def generate_changelog(build: 'models.Build') -> typing.Optional[str]:
         return None
 
     # Grab the RPM header of the previous update, and generate a ChangeLog
-    oldh = get_rpm_header(lastpkg)
-    oldtime = oldh['changelogtime']
-    text = oldh['changelogtext']
-
-    if not text:
-        oldtime = 0
-    elif isinstance(oldtime, list):
-        oldtime = oldtime[0]
+    oldtime = 0
+    if lastpkg is not None:
+        oldh = get_rpm_header(lastpkg)
+        if oldh['changelogtext']:
+            oldtime = oldh['changelogtime']
+            if isinstance(oldtime, list):
+                oldtime = oldtime[0]
 
     return build.get_changelog(oldtime)
 
```
It's basically the same except I only overwrite oldtime with changelogtime if it's necessary. A cleaner solution would be to have a small `_getoldtime()` function, but eh :-)